src/linear_algebra/affine_space/combination.lean

```diff
@@ -359,6 +378,83 @@ rfl
   ({i} : finset ι).centroid k p = p i :=
 by simp [centroid_def, affine_combination_apply]
 
+/-- The centroid of two points. -/
+lemma centroid_insert_singleton [char_zero k] (p : ι → P) (i₁ i₂ : ι) :
```

Comment:
Instead of `[char_zero k]` I guess you can also ask for `[invertible (2 : k)]`, right?

Reply:
I've changed those two lemmas to use `invertible`. (Note that it's still the case that all `centroid` definitions and lemmas require a `division_ring`, while `invertible` applies to more general rings.)